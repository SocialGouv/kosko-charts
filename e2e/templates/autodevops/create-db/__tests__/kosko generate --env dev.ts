//

import { config } from "dotenv";
import { KOSKO_BIN, template, TIMEOUT } from "e2e/templates/helpers";
import execa from "execa";
import { basename, resolve } from "path";

//

const cwd = template(basename(resolve(__dirname, "..", "..")));

test(
  "create DB from gitlab : kosko generate jobs/create-db --env dev",
  async () => {
    const gitlabEnv = config({
      path: resolve(cwd, "./environments/.gitlab-ci.env"),
    }).parsed;

    const env = {
      ...gitlabEnv,
    };

    // Required to allow seemless integration code example
    const result = await execa.node(
      KOSKO_BIN,
      ["generate", "jobs/create-db", "--env", "dev"],
      {
        cwd,
        env,
      }
    );

    expect(result.stdout).toMatchSnapshot();
    expect(result.exitCode).toEqual(0);
  },
  TIMEOUT
);

test(
  "create DB from github : kosko generate jobs/create-db --env dev",
  async () => {
    const gitlabEnv = config({
      path: resolve(cwd, "./environments/.github-actions.env"),
    }).parsed;

    const env = {
      ...gitlabEnv,
      SOCIALGOUV_CONFIG_PATH: __dirname + "/config.json",
    };

    // Required to allow seemless integration code example
    const result = await execa.node(
      KOSKO_BIN,
      ["generate", "jobs/create-db", "--env", "dev"],
      {
        cwd,
        env,
      }
    );

    expect(result.stdout).toMatchSnapshot();
    expect(result.exitCode).toEqual(0);
  },
  TIMEOUT
);
