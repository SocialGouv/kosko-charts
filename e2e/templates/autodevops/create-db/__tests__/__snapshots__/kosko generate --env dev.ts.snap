// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create DB from github : kosko generate jobs/create-db --env dev 1`] = `
"apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    app.github.com/job: xxxxxxx-job
    app.github.com/ref: refs/heads/e2e-branch
    app.github.com/repo: socialgouv/sample-kosko
    app.github.com/run: '12345'
    app.github.com/sha: '8843083'
  labels:
    application: sample-kosko
    component: sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
  name: create-db-user
  namespace: sample-kosko-e2e-branch
spec:
  backoffLimit: 5
  template:
    spec:
      containers:
        - command:
            - create-db-user
          env:
            - name: NEW_DB_NAME
              valueFrom:
                secretKeyRef:
                  key: PGDATABASE
                  name: azure-pg-user-e2e-branch
            - name: NEW_USER
              valueFrom:
                secretKeyRef:
                  key: PGUSER
                  name: azure-pg-user-e2e-branch
            - name: NEW_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: PGPASSWORD
                  name: azure-pg-user-e2e-branch
            - name: PGHOST
              valueFrom:
                secretKeyRef:
                  key: PGHOST
                  name: azure-pg-user-e2e-branch
            - name: NEW_DB_EXTENSIONS
              value: hstore pgcrypto citext uuid-ossp
          envFrom:
            - secretRef:
                name: azure-pg-admin-user
          image: ghcr.io/socialgouv/docker/azure-db:6.64.0
          imagePullPolicy: IfNotPresent
          name: create-db-user
          resources:
            limits:
              cpu: 300m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 64Mi
      restartPolicy: Never
  ttlSecondsAfterFinished: 86400"
`;

exports[`create DB from gitlab : kosko generate jobs/create-db --env dev 1`] = `
"apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-sample-kosko
    app.gitlab.com/env: e2e-branch-42
    app.gitlab.com/env.name: e2e-branch-dev2
  labels:
    application: e2e-branch-42-sample-kosko
    component: e2e-branch-42-sample-kosko
    owner: sample-kosko
    team: sample-kosko
    cert: wildcard
  name: create-db-user
  namespace: sample-kosko-24-e2e-branch-42
spec:
  backoffLimit: 5
  template:
    spec:
      containers:
        - command:
            - create-db-user
          env:
            - name: NEW_DB_NAME
              valueFrom:
                secretKeyRef:
                  key: PGDATABASE
                  name: azure-pg-user-e2e-branch
            - name: NEW_USER
              valueFrom:
                secretKeyRef:
                  key: PGUSER
                  name: azure-pg-user-e2e-branch
            - name: NEW_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: PGPASSWORD
                  name: azure-pg-user-e2e-branch
            - name: PGHOST
              valueFrom:
                secretKeyRef:
                  key: PGHOST
                  name: azure-pg-user-e2e-branch
            - name: NEW_DB_EXTENSIONS
              value: hstore pgcrypto citext uuid-ossp
          envFrom:
            - secretRef:
                name: azure-pg-admin-user
          image: ghcr.io/socialgouv/docker/azure-db:6.64.0
          imagePullPolicy: IfNotPresent
          name: create-db-user
          resources:
            limits:
              cpu: 300m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 64Mi
      restartPolicy: Never
  ttlSecondsAfterFinished: 86400"
`;
